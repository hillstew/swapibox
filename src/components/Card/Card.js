import React from "react";

export const Card = props => {
  let peopleCard = (
    <div>
      <p>Name: {props.subject.name}</p>
      <p>Homeworld: {props.subject.homeworld}</p>
      <p>Species: {props.subject.species}</p>
      <p>Population: {props.subject.population}</p>
      <p>Favorite: {props.subject.favorite}</p>
    </div>
  );

  let planetCard = (
    <div>
      <p>Name: {props.subject.name}</p>
      <p>Terrain: {props.subject.terrain}</p>
      <p>Population: {props.subject.population}</p>
      <p>Climate: {props.subject.climate}</p>
      <p>Residents: {props.subject.residents}</p>
      <p>Favorite: {props.subject.favorite}</p>
    </div>
  );

  let vehicleCard = (
    <div>
      <p>Name: {props.subject.name}</p>
      <p>Model: {props.subject.model}</p>
      <p>Class: {props.subject.class}</p>
      <p>Passengers: {props.subject.passengers}</p>
      <p>Favorite: {props.subject.favorite}</p>
    </div>
  );

  let showFavorites = () => {

  }

  return (
    <div className="card">
      {props.view === "vehicles" && <div>{vehicleCard}</div>}
      {props.view === "planets" && <div>{planetCard}</div>}
      {props.view === "people" && <div>{peopleCard}</div>}
      {/* {props.view === "favorites" && {showFavorites}} */}
    </div>
  );
};
