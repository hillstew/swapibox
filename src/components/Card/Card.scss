.card {
  background-color: rgb(0, 0, 29);
  border: 5px solid #cac6c6;
  height: 250px;
  margin: 20px auto;
  width: 400px;
  
  p {
    color: #ffffffbd;
    font-size: 1.2rem;
    padding-left: 15px;
    text-align: left;
  }
}