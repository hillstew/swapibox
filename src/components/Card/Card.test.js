import React from "react";
import Card from "../Card/Card";
import { shallow } from "enzyme";

const mockSubject = {
  name: "T-16 skyhopper",
  model: "T-16 skyhopper",
  class: "repulsorcraft",
  passengers: "1"
};

const mockView = "vehicles";
const mockKey = mockSubject.name;

describe("Card", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Card subject={mockSubject} view={mockView} key={mockKey} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
