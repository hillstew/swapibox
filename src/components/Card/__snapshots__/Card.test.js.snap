// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="card"
>
  <div>
    <div>
      <p>
        Name: 
        T-16 skyhopper
      </p>
      <p>
        Model: 
        T-16 skyhopper
      </p>
      <p>
        Class: 
        repulsorcraft
      </p>
      <p>
        Passengers: 
        1
      </p>
    </div>
  </div>
</div>
`;
